### PR TITLE
Add a better support for package signing

### PR DESCRIPTION
**Pull request by adriendelsalle**
Description
---

Add endpoints to post trust roles
Add database models
Add tests
Refactor reposigner
Support head requests
Add a post_index_creation hook
Use post_index_creation to sign artifacts
Use tmp_path for generated test data
Add alembic migration

### CODE REVIEW

File: quetz/jobs/models.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ class Job(Base):
     id = sa.Column(sa.Integer, primary_key=True)
     created = sa.Column(sa.DateTime, nullable=False, default=datetime.utcnow)
     updated = sa.Column(sa.DateTime, nullable=False, default=datetime.utcnow)
-    manifest = sa.Column(sa.Binary(), nullable=False)
+    manifest = sa.Column(sa.LargeBinary(), nullable=False)
     owner_id = sa.Column(
```
Comment (wolfv) [inline quetz/jobs/models.py:45]: Do we need this change here? Might deserve a dedicated PR, no?

File: quetz/jobs/runner.py [hunk 2]
```diff
@@ -186,7 +186,8 @@ def run_jobs(self, job_id=None, force=False):
         now = datetime.utcnow()
         db = self.db
         jobs = db.query(Job).filter(Job.status == JobStatus.pending)
-        logger.info(f"Got pending jobs: {jobs.count()}")
+        if jobs.count():
```
Comment (wolfv) [inline quetz/jobs/runner.py:189]: this is unrelated

File: quetz/migrations/versions/794249a0b1bd_adding_jobs_tables.py [hunk 3]
```diff
@@ -22,7 +22,7 @@ def upgrade():
         sa.Column('id', sa.Integer(), nullable=False),
         sa.Column('created', sa.DateTime(), nullable=False),
         sa.Column('updated', sa.DateTime(), nullable=False),
-        sa.Column('manifest', sa.Binary(), nullable=False),
+        sa.Column('manifest', sa.LargeBinary(), nullable=False),
```
Comment (wolfv) [inline quetz/migrations/versions/794249a0b1bd_adding_jobs_tables.py:25]: we should never change a historic migration

File: quetz/tasks/indexing.py [hunk 4]
```diff
@@ -265,7 +272,7 @@ def update_indexes(dao, pkgstore, channel_name, subdirs=None):
         after_upload_move.append(dest)
 
     for f_to_move in after_upload_move:
-        logger.info(
+        logger.debug(
```
Comment (wolfv) [inline quetz/tasks/indexing.py:275]: unrelated change

**Comment by adriendelsalle** [comment]
@wolfv this is a draft PR, not ready for review. I'm splitting some changes in dedicated PRs

**Comment by wolfv** [comment]
yep, no worries, i just gave some very quick feedback :)

File: quetz/jobs/models.py (hunk 1 above)
Comment (adriendelsalle) [reply quetz/jobs/models.py:45]: sure, see https://github.com/mamba-org/quetz/pull/474

**Comment by adriendelsalle** [comment]
It looks like I'm missing something about declaring plugin deps. `mamba` is probably installed from `pip` leading to wrong package installed and failing imports.
@wolfv do you have any idea?

**Comment by wolfv** [comment]
Now you just need to run the linters! :)
